compiler/plugins/target/ROCM/Dialect/ROCM/Transforms/Passes.td

```diff
@@ -30,4 +30,29 @@ def ApplyBuiltinPDLPatternsPass : Pass<"iree-rocm-apply-builtin-pdl-patterns", "
   ];
 }
 
+
+def ApplyBuiltinPDLPatternsDriverPass : Pass<"iree-rocm-apply-builtin-pdl-patterns-driver", "mlir::ModuleOp"> {
+  let summary = "Applies builtin PDL patterns on each function-like op within a module";
+  let description = [{
+    Parses a builtin file containing specialization and/or PDL patterns based on the target arch and applies them as patterns
+  }];
+  let options = [
+    Option<"enableSpecialization", "enable-specialization", "bool",
+           /*defaultValue=*/"false",
+           "Enable runtime specialization of dynamically shaped dispatches">,
+    Option<"enableTensorUkernels", "enable-tensor-ukernels", "bool",
+           /*defaultValue=*/"false", "Enable MLIR-tensor-based ukernels">
+  ];
+  let dependentDialects = [
+    "iree_compiler::IREE::HAL::HALDialect",
+    "iree_compiler::IREE::ROCM::ROCMDialect",
+    "iree_compiler::IREE::Codegen::IREECodegenDialect",
+    "mlir::gpu::GPUDialect",
+    "mlir::pdl::PDLDialect",
+    "mlir::pdl_interp::PDLInterpDialect",
+    "mlir::vector::VectorDialect",
+    "ROCDL::ROCDLDialect",
```

Comment:
I should have added this for the `ApplyBuiltinPDLPatterns` pass too, but a short comment explaining the dialect dependencies would be good. Something like
```
// ROCDL, GPU, and Vector dialect dependencies are to make sure these dialects are loaded when parsing
// the ukernels that contain them. The pass itself does not depend directly on these dialects.
```

Reply:
Thanks, I added this!